src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,196 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head = null;
+    private Node<T> tail = null;
+    private int size = 0;
+
     @Override
     public void add(T value) {
+        addTail(value);
     }
 
     @Override
     public void add(T value, int index) {
+        checkOnIndexOutOfBoundsException(index, false);
+
+        if (size == 0) {
+            addHead(value);
+        } else if (size == index) {
+            addTail(value);
+        } else {
+            Node<T> findedNode = findNode(index);
+            addBefore(value, findedNode);
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            addTail(value);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return findNodeAndCheckIndex(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> node = findNodeAndCheckIndex(index);
+        T oldValue = node.value;
+        node.value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> node = findNodeAndCheckIndex(index);
+        T removeValue = node.value;
+        removeNode(node);
+        return removeValue;
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        boolean isNode = false;
+        Node<T> node = head;
+        if (size != 0) {
+            do {
+                T value = node.value;
+                if ((value == null && object == null)
+                        || (value != null && node.value.equals(object))) {
+                    removeNode(node);
+                    isNode = true;
+                    break;
+                } else {
+                    node = node.next;
+                }
+            } while (node.next != null);
+        }
```

Comment:
```suggestion
        while (node != null) {
            if ((value == null && object == null)
                        || (value != null && node.value.equals(object))) {
                    removeNode(node);
                    return true;
            }
            node = node.next;
        }
```

Reply:
Thank you a lot, i look at the solution for Remove method from other side